catalog/models.py

```diff
@@ -1,3 +1,124 @@
+""" catalog models for data storage
+
+The models are defined as classes for every real world object needed in
+the system. Each model class needs to inherid from django.db.models.Model.
+The methods for CRUD (create, read, update and delete) are inherited from
+the super class.
+"""
+
+
 from django.db import models
+import uuid
+from django.db.models.deletion import SET_NULL
+from django.urls import reverse     # Used to generate URLs by reversing the URL patterns
+
+class Genre(models.Model):
+    """Model representing a book genre."""
+
+    name = models.CharField(max_length=200, help_text='Enter a book genre (e.g. Science Fiction)')
+
+    def __str__(self):
+        """String for representing the Model object."""
+        return self.name
+
+class Language(models.Model):
+    """Model representing the language for a book."""
+
+    language = models.CharField(max_length=100, help_text='Enter a language for the book (e.g. English)')
+
+
+    class Meta:
+        ordering = ['language']
+
+
+    def __str__(self):
+        """String for representing the Model object."""
+        return self.language
+
+
+class Book(models.Model):
+    """Model representing a book (but not a specific copy of a book)."""
+    title = models.CharField(max_length=200)
+
+    # Foreign Key used because book can only have one author, but authors can have multiple books
+    # Author as a string rather than object because it hasn't been declared yet in the file
+    author = models.ForeignKey('Author', on_delete=models.SET_NULL, null=True)
+
+    summary = models.TextField(max_length=1000, help_text='Enter a brief description of the book')
+    isbn = models.CharField('ISBN', max_length=13, unique=True,
+                             help_text='13 Character <a href="https://www.isbn-international.org/content/what-isbn">ISBN number</a>')
+
+    # ManyToManyField used because genre can contain many books. Books can cover many genres.
+    # Genre class has already been defined so we can specify the object above.
+    genre = models.ManyToManyField(Genre, help_text='Select a genre for this book')
+
+
+    # The language is stored as a OneToOneField because a book can have only one language
+    # The Language class is already defined, so it can be used to define the relation
+    language = models.ForeignKey('Language', Language, null = True)
```

Comment:
Question: Both `Language` as string and class are passed as args to this `ForeignKey` function. Why is that?

Reply:
Same as the issue with Author. I didn't understand the syntax properly before. I have changed it to class only now.